core/src/main/java/smashdudes/ecs/systems/CountdownCameraSystem.java

```diff
@@ -0,0 +1,79 @@
+package smashdudes.ecs.systems;
+
+import com.badlogic.gdx.graphics.Camera;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.math.Vector3;
+import com.badlogic.gdx.utils.Array;
+import smashdudes.ecs.Engine;
+import smashdudes.ecs.Entity;
+import smashdudes.ecs.components.*;
+
+import java.lang.reflect.GenericArrayType;
+
+public class CountdownCameraSystem extends GameSystem
+{
+    private final Vector2 position = new Vector2(0, 0);
+    private int numPlayers;
+    private int currPlayer;
+
+    private float timer;
+
+    public CountdownCameraSystem(Engine engine)
+    {
+        super(engine);
+
+        registerComponentType(CountdownComponent.class);
+        registerComponentType(CameraComponent.class);
+
+        numPlayers = 0;
+        currPlayer = 1;
+        timer = 0;
+    }
+
+    @Override
+    public void preUpdate()
+    {
+        Array<Entity> entities = engine.getEntities(PositionComponent.class, PlayerComponent.class);
+        for(Entity e : entities)
+        {
+            int ID = e.getComponent(PlayerComponent.class).handle.ID;
+
+            if(ID == currPlayer)
+            {
+                position.set(e.getComponent(PositionComponent.class).position);
+            }
+            if(ID > numPlayers)
+            {
+                numPlayers = ID;
```

Comment:
This is incorrect and is just coincidence if it works. numPlayers should be equal to entities.size

Reply:
the statement above it should probably be changed as well then